src/components/probingDataAlert/ProbingDataAlert.tsx

```diff
@@ -0,0 +1,48 @@
+import { Alert, Grid } from '@mui/material'
+import { useTranslation } from 'react-i18next'
+import { useState, useEffect } from 'react'
+
+interface ProbingData {
+  probingEnabled: boolean
+  state: string
+  eserviceActive: boolean
+}
+
+export const ProbingDataAlert: React.FC<ProbingData> = ({
+  probingEnabled,
+  state,
+  eserviceActive,
+}) => {
+  const [message, setMessage] = useState('')
+
+  const { t } = useTranslation(['detailsPage'])
+
+  useEffect(() => {
+    if (probingEnabled === false && state === 'n/d') {
+      setMessage('monitoringSystemSuspendedMessage')
+    } else if (state === 'offline' && !eserviceActive) {
+      setMessage('versionSuspendedMessage')
+    } else if (state === 'offline' && eserviceActive) {
+      setMessage('eserviceNotAnswerMessage')
+    } else {
+      setMessage('')
+    }
+  }, [probingEnabled, state, eserviceActive])
+
+  return (
+    <>
+      {message !== '' ? (
+        <Grid item container>
+          <Alert
+            sx={{
+              width: '100%',
+            }}
+            severity="warning"
+          >
+            {t(message)}
+          </Alert>
+        </Grid>
+      ) : null}
+    </>
+  )
+}
```

Comment:
Il valore di `message` può essere derivato direttamente dalle props senza dover aspettare il trigger del `useEffect` per la sincronizzazione con lo state.

In generale, quando un valore può essere derivato dalle props o dallo state, è sempre meglio computarlo durante il render. Limitare l'utilizzo di `useEffect` alla sola sincronizzazione con risorse "esterne" al lyfecycle di React (ex: chiamate API, eventi del DOM...). 

A tal proposito, consiglio [questo articolo](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes) scritto da un membro del core team di React.

```suggestion
export const ProbingDataAlert: React.FC<ProbingData> = ({
  probingEnabled,
  state,
  eserviceActive,
}) => {
  const { t } = useTranslation(['detailsPage'])  

  let message = ""

  if (probingEnabled === false && state === 'n/d') {
    message = 'monitoringSystemSuspendedMessage'
  } else if (state === 'offline' && !eserviceActive) {
    message = 'versionSuspendedMessage'
  } else if (state === 'offline' && eserviceActive) {
    message = 'eserviceNotAnswerMessage'
  }

  if (!message) return null

  return (
    <Grid item container>
      <Alert
        sx={{
          width: '100%',
        }}
        severity="warning"
      >
        {t(message)}
      </Alert>
    </Grid>
  )
}
```